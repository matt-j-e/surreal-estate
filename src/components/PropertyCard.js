import React from "react";

const PropertyCard = ({
  title,
  type,
  bedrooms,
  bathrooms,
  price,
  city,
  email,
}) => {
  return (
    <article className="property-card">
      <div className="property-card__title">{title}</div>
      <div className="property-card__type">{type}</div>
      <div className="property-card__bedrooms">{bedrooms}</div>
      <div className="property-card__bathrooms">{bathrooms}</div>
      <div className="property-card__price">{price}</div>
      <div className="property-card__city">{city}</div>
      <div className="property-card__email">{email}</div>
    </article>
  );
};

export default PropertyCard;
