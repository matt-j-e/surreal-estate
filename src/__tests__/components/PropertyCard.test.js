import React from "react";
import { render } from "@testing-library/react";
import PropertyCard from "../../components/PropertyCard";

describe("PropertyCard", () => {
  const validProps = {
    title: "3 bedroom house",
    type: "Semi-Detached",
    bedrooms: "3",
    bathrooms: "1",
    price: "200000",
    city: "Manchester",
    email: "name@email.com",
  };

  it("renders correctly", () => {
    const { asFragment } = render(
      <PropertyCard
        title={validProps.title}
        type={validProps.type}
        bedrooms={validProps.bedrooms}
        bathrooms={validProps.bathrooms}
        price={validProps.price}
        city={validProps.city}
        email={validProps.email}
      />
    );
    expect(asFragment()).toMatchSnapshot();
  });

  it("renders correct values for props", () => {
    const { getByText } = render(
      <PropertyCard
        title={validProps.title}
        type={validProps.type}
        bedrooms={validProps.bedrooms}
        bathrooms={validProps.bathrooms}
        price={validProps.price}
        city={validProps.city}
        email={validProps.email}
      />
    );
    expect(getByText("3 bedroom house")).toHaveClass("property-card__title");
    expect(getByText(/semi-detached/i)).toHaveClass("property-card__type");
    expect(getByText("3")).toHaveClass("property-card__bedrooms");
    expect(getByText("1")).toHaveClass("property-card__bathrooms");
    expect(getByText("200000")).toHaveClass("property-card__price");
    expect(getByText("Manchester")).toHaveClass("property-card__city");
    expect(getByText("name@email.com")).toHaveClass("property-card__email");
  });
});
