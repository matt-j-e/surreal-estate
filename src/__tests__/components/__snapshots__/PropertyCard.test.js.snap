// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PropertyCard renders correctly 1`] = `
<DocumentFragment>
  <article
    class="property-card"
  >
    <div
      class="property-card__title"
    >
      3 bedroom house
    </div>
    <div
      class="property-card__type"
    >
      Semi-Detached
    </div>
    <div
      class="property-card__bedrooms"
    >
      3
    </div>
    <div
      class="property-card__bathrooms"
    >
      1
    </div>
    <div
      class="property-card__price"
    >
      200000
    </div>
    <div
      class="property-card__city"
    >
      Manchester
    </div>
    <div
      class="property-card__email"
    >
      name@email.com
    </div>
  </article>
</DocumentFragment>
`;
